tools/query_verification/src/main/java/com/google/bigquery/BigQueryManager.java

```diff
@@ -279,8 +281,55 @@ public String getName() {
             StandardSQLTypeName type = fields.get(i).getType().getStandardType();
 
             Object result;
-            // TODO Convert value to appropriate Java type
-            result = value.getStringValue();
+            try {
+                switch (type) {
+                    case BOOL:
+                        // returns boolean
+                        result = value.getBooleanValue();
+                        break;
+                    case FLOAT64:
+                        // returns double
+                        result = value.getDoubleValue();
+                        break;
+                    case INT64:
+                        // returns long
+                        result = value.getLongValue();
+                        break;
+                    case NUMERIC:
+                        // returns BigDecimal
+                        result = value.getNumericValue();
+                        break;
+                    case STRUCT:
+                        FieldList subFields = fields.get(i).getSubFields();
+                        FieldValueList subValues = value.getRecordValue();
+                        List<Object> subResults = parseResults(subValues, subFields);
```

Comment:
I think you can combine this line like `result = parseResult(subValues, subFields)`.